src/ui/public/filter_editor/params_editor/filter_params_values_controller.js

```diff
@@ -0,0 +1,32 @@
+
+
+import _ from 'lodash';
+import chrome from 'ui/chrome';
+
+const baseUrl = chrome.addBasePath('/api/kibana/values');
+
+export function filterParamsValuesController($http, $scope, $rootScope) {
+
+  this.getAllFieldValues = getAllFieldValues;
+
+
+  function getAllFieldValues(field) {
+
+    field = $scope.field;
+    if (!_.get(field, 'aggregatable') || field.type !== 'string') {
+      return Promise.resolve([]);
+    }
+
+    const postParams = {
+      field: field.name,
+      query: $rootScope.filterState
```

Comment:
Rather than writing the filter state to `$rootScope`, maybe we can use a temporary `SearchSource` to determine the current filters? @elastic/kibana-discovery? Might work something like this:

```js
export function filterParamsValuesController($http, $scope, courier) {
 
  this.getAllFieldValues = function (field) {
    field = $scope.field;

    if (!_.get(field, 'aggregatable') || field.type !== 'string') {
      return Promise.resolve([]);
    }
 	
	const searchSource = new courier.SearchSource({
      index: field.indexPattern
    })

    return searchSource.getSearchRequestBody()
      .then(({ query }) => (
        $http.post(`${baseUrl}/${field.indexPattern.title}`, {
          field: field.name,
          query
        })
      ))
      .then(response => {
        $scope.$emit('value_event', response.data);
      });
  }

}
```

Reply:
btw, these are just ideas, not tested or sure it works well in all situations :)

@Stacey-Gammon do you know why we don't set the search source in apps like Discover to be the root search source?

Reply:
Ah, no, but I don't even like that dashboard does it.  It's created some strange bugs.  See https://github.com/elastic/kibana/pull/17123 and https://github.com/elastic/kibana/issues/16641 - we are workin towards getting rid of the search source inheritance all together.
Instead we want dashboard to pass filters, time range and query explicitly down to it's embeddables.

I haven't looked really at any of this PR but I'd be worried that any reliance on searchSource inheritance will break in the future.

Reply:
This could probably just pull in the `queryFilter` and `timeFilter` services, I'm mostly just worried about not doing things the same way they are done by the app itself that might cause the values retrieved by this filter to not match the values being rendered by the vis/discover/dashboard they are supposed to match.

Reply:
Haven't had much time to look into this either, but just want to second Stacey here. We will remove search source inheritance rather sooner than later (hopefully within one of the next minor versions). I really think we shouldn't build anything new that relies on that anymore. And of course also second, not introducing any new Angular code here, since we are really hardly trying to get rid of Angular.